pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -821,3 +1160,23 @@ func checkNodeAffinity(og *operationGenerator, volumeToMount VolumeToMount, plug
 	}
 	return nil
 }
+
+// isDeviceOpened checks the device status if the device is in use anywhere else on the system
+func isDeviceOpened(deviceToDetach AttachedVolume, mounter mount.Interface) (bool, error) {
+	isDevicePath, devicePathErr := mounter.PathIsDevice(deviceToDetach.DevicePath)
+	var deviceOpened bool
+	var deviceOpenedErr error
+	if !isDevicePath && devicePathErr == nil ||
+		(devicePathErr != nil && strings.Contains(devicePathErr.Error(), "does not exist")) {
+		// not a device path or path doesn't exist
+		//TODO: refer to #36092
+		glog.V(3).Infof("The path isn't device path or doesn't exist. Skip checking device path: %s", deviceToDetach.DevicePath)
+		deviceOpened = false
+	} else {
+		deviceOpened, deviceOpenedErr = mounter.DeviceOpened(deviceToDetach.DevicePath)
+		if deviceOpenedErr != nil {
+			return false, deviceToDetach.GenerateErrorDetailed("UnmountDevice.DeviceOpened failed", deviceOpenedErr)
```

Comment:
This "UnmountDevice" error is also going to return for block?

Reply:
Should remove UnmountDevice. tag since this method is used both UnmountDevice and UnmapDevice.